added parsing csv

TatianaPirch opened the pull request:
1. added dependencies for lombok and log4j
	2. added log4j.properties
	3. added entity (for testing)
	4. added ReviewRepository (for testing)
	5. added ServiceParsing
	6. added DataInsertController
	7. added log_file.log

TatianaPirch committed:
added parsing csv
	1. added dependencies for lombok and log4j
	2. added log4j.properties
	3. added entity (for testing)
	4. added ReviewRepository (for testing)
	5. added ServiceParsing
	6. added DataInsertController
	7. added log_file.log

boroda4436 commented on log/log_file.log:
```diff
@@ -0,0 +1,8 @@
+
+
+2019-11-06 14:25:18 INFO  ServiceParsing:69 - Read number of items: 100000
```
@TatianaPirch should we push it?

TatianaPirch replied:
I attached a log file to show how long it took to process csv-file

boroda4436 commented on src/main/java/team/five/amazonreviewutil/controller/DataInsertController.java:
```diff
@@ -0,0 +1,22 @@
+@Component
+public class DataInsertController {
+
+    private static final String DATA_FILE_NAME = "D:\\My\\MA\\project\\Reviews.csv";
```
@TatianaPirch what if I will run this on my Mac? Will it works?

SkaterWeather replied:
It's will not on work in another Windows(like mine or Kolya's) too;
Idk, i tried to see reliable path file in idea, but it don't worked for me;
@TatianaPirch please place .csv file in package src/main/resources/ and push it in branch too;

SkaterWeather replied:
String txt = "./src/resources/Reviews.csv" , try it out, maybe will work for you

boroda4436 commented on src/main/java/team/five/amazonreviewutil/repositiry/ReviewRepository.java:
```diff
@@ -0,0 +1,7 @@
+import team.five.amazonreviewutil.entity.Review;
+
+public interface ReviewRepository extends JpaRepository<Review, Long> {
+}
```
```suggestion
}

```

boroda4436 commented on src/main/java/team/five/amazonreviewutil/service/ServiceParsing.java:
```diff
@@ -0,0 +1,82 @@
+    private final String charsetName = "UTF-8";
+    Charset charset = Charset.forName(charsetName);
+    private final char lineBreak = '\n';
+    private static final String DATA_FILE_NAME = "D:\\My\\MA\\project\\test2.csv";
```
@NikolaiKatsai  @SkaterWeather  can you run this on your PC? Is everything ok?

SkaterWeather replied:
Same, i responsed

boroda4436 commented on src/main/java/team/five/amazonreviewutil/service/ServiceParsing.java:
```diff
@@ -0,0 +1,82 @@
+    private int itemsRead;
+    private final String fieldSeparator = ",";
+    private final String charsetName = "UTF-8";
+    Charset charset = Charset.forName(charsetName);
```
```suggestion
    Charset charset = StandardCharsets.UTF_8;
```

boroda4436 commented on src/main/java/team/five/amazonreviewutil/service/ServiceParsing.java:
```diff
@@ -0,0 +1,82 @@
+        serviceParsing.processInputFile(DATA_FILE_NAME);
+    }*/
+
+    public void processInputFile(String dataFileName) throws IOException {
```
@TatianaPirch How long it takes to process the file?

TatianaPirch replied:
about 2 minutes

TatianaPirch committed:
1.did not add the Reviews.csv , it is too big (Reviews.csv should be in the folder resources)
2.fixed problem with path
3.renamed parser
4.added check style
5.added .log to gitignore

TatianaPirch committed:
valid commit
1.did not add the Reviews.csv , it is too big (Reviews.csv should be in the folder resources)
2.fixed problem with path
3.renamed parser
4.added check style
5.added .log to gitignore

boroda4436 commented on src/main/java/team/five/amazonreviewutil/service/CsvParserService.java:
```diff
@@ -0,0 +1,82 @@
+    Charset charset = StandardCharsets.UTF_8;
+    private final char lineBreak = '\n';
+
+    public void processInputFile(String fileName) throws IOException {
```
@TatianaPirch Isn't it too much for one method??? Did you consider to separate the functionality?

TatianaPirch committed:
1.added two methods saveReviewsToDb and createReview
2.multithreading is planned to be implemented in future.
 need a separate ticket for add multithreading
